flang/lib/Lower/CharacterRuntime.cpp

```diff
@@ -46,6 +46,33 @@ static int discoverKind(mlir::Type ty) {
 // Lower character operations
 //===----------------------------------------------------------------------===//
 
+
+void Fortran::lower::genAdjustL(Fortran::lower::FirOpBuilder &builder,
+                                mlir::Location loc, mlir::Value resultBox,
+                                mlir::Value stringBox) {
+  auto adjustLFunc = getRuntimeFunc<mkRTKey(Adjustl)>(loc, builder);
+  auto fTy = adjustLFunc.getType();
+  auto sourceFile = Fortran::lower::locationToFilename(builder, loc);
+  auto sourceLine =
+      Fortran::lower::locationToLineNo(builder, loc, fTy.getInput(3));
+  auto args = Fortran::lower::createArguments(builder, loc, fTy, resultBox,
+                                              stringBox, sourceFile, sourceLine);
+  builder.create<fir::CallOp>(loc, adjustLFunc, args);
+}
+
+void Fortran::lower::genAdjustR(Fortran::lower::FirOpBuilder &builder,
+                                mlir::Location loc, mlir::Value resultBox,
+                                mlir::Value stringBox) {
+  auto adjustRFunc = getRuntimeFunc<mkRTKey(Adjustr)>(loc, builder);
```

Comment:
These two functions are mostly duplicate save for the `getRuntimeFunc` call and result. It should be possible then to merge them into a common codegen function with the result of the `getRuntimeFunc` call as 1 additional parameter, right?

Reply:
I tried that but had some problems getting it fly -- in-between meetings... ;-) -- let me give it a go again.  Stay tuned...

Reply:
Just pushed this suggested refactoring.  Take a look at let me know your thoughts.  Will squash, etc. once I hear back from you.